Add-map-page

DavidS252 opened the pull request:
Made a Page for the map display of the app. the whole page contains 3/4th of a map on the right side and 1/4 of a panel in beautiful teal color to display the information about the map. 
Later on we could decide how big the panel on the left will be and what information it will display. for now the route for the map page is /map.

Commit:
Added the pages to showcase the map on the page of /map

Commit:
forgot to add

Commit:
made a beautiful map with a panel on the left to write description and etc.

DavidS252 commented:
![Screenshot 2025-03-31 210845](https://github.com/user-attachments/assets/478b2149-0f0c-43a5-937d-76cf70c09349)

this is the screenshot for the map page. for now it mostly default but later on on the project it will be full of tiles and vectors to showcase specific locations with vectoring and geojson.

Ilana-Belokon2 commented on src/components/Map/MapPage.tsx:
```diff
@@ -0,0 +1,45 @@
+    map.current = new maptilersdk.Map({
+      container: mapContainer.current,
+      style: maptilersdk.MapStyle.STREETS,
+      center: [rishonLezion.lng, rishonLezion.lat],
```
you need to display coordinates and icon for all the real estates, not just random Rishon Lezion
get the real estate data from the server(from the db) and mark it on the map

DavidS252 replied:
i did it, as the genius geocoding that i am. be careful though because the api we use is limited and we have around 1000 requests/month

Ilana-Belokon2 commented on src/components/Map/MapPage.tsx:
```diff
@@ -0,0 +1,45 @@
+    <div className="map-page-container">
+      <div className="info-panel">
+        <h2>Info Panel</h2>
+        <p>This area can contain details about the map, location, etc.</p>
```
write the details of the real estates displayed on the map.
when i am clicking on one real estate that mark on the map than i need to see its details on the left side

DavidS252 replied:
made the appropriate changes lol

Ilana-Belokon2 reviewed (changes requested):
looks good, need to take care of the real estates

Ilana-Belokon2 commented:
use material library for the design, instead or in addition to pure css

Commit:
Merge branch 'dev' into add-map-page

Commit:
added the possibility to fetch all the properties and show them on the map

DavidS252 commented:
fixed what ilana told me to fix and added new features into the page itself
![image](https://github.com/user-attachments/assets/0f26177c-6108-4606-8d96-259f13358b83)

DavidS252 commented:
next thing to do:
1. design it to look better (to look as the mockup + a bit of ui/ux wouldn't hurt here)
2. I didn't made for now the map-service.tsx file as for how the api works. i'll do it in another branch with some fixes i want to do to my code. this branch an PR is to start out the 'Map Page'.

Ilana-Belokon2 commented on src/components/Map/MapPage.tsx:
```diff
@@ -0,0 +1,136 @@
+import { MAPTILER_API_KEY } from '../../config';
+import RealEstateService  from '../../services/realestate-service';  // Service to fetch real estate data
+
+// Define the shape of a real estate listing (if not already defined/imported)
```
delete all chat comments

Commit:
made removal of comments in the code for more aesthetics

Ilana-Belokon2 commented:
look at the design of the ai-recommensations or the profile and use similar design for the left side of the page(font, colors).
use material library for pure design.

Commit:
created seperate file for map service

Commit:
changes in the css

Ilana-Belokon2 commented:
looks very good. maybe fix the buttons according to mockup, and add the name of the real estate, not just the address because you cant know what is the real estate

Commit:
made more beautiful error handling and more error info for the clients.

DavidS252 commented:
> looks very good. maybe fix the buttons according to mockup, and add the name of the real estate, not just the address because you cant know what is the real estate

the buttons will be integrated later on on the project to look more like the mockup + the mockup a bit ugly than the real thing we are building.
Though i added the error handling and showing some information to the client so that they will know what's happening.

Another thing there is no name for the real estate, but i added the owner and the area to showcase beautifully on the buttons.

morielnoam committed:
Merge branch 'dev' into add-map-page
